proxy/src/test/java/com/wavefront/agent/PreprocessorRulesTest.java

```diff
@@ -0,0 +1,274 @@
+package com.wavefront.agent;
+
+import com.google.common.collect.Lists;
+
+import com.wavefront.agent.preprocessor.AgentPreprocessorConfiguration;
+import com.wavefront.agent.preprocessor.PointLineBlacklistRegexFilter;
+import com.wavefront.agent.preprocessor.PointLineReplaceRegexTransformer;
+import com.wavefront.agent.preprocessor.PointLineWhitelistRegexFilter;
+import com.wavefront.agent.preprocessor.ReportPointAddPrefixTransformer;
+import com.wavefront.agent.preprocessor.ReportPointAddTagIfNotExistsTransformer;
+import com.wavefront.agent.preprocessor.ReportPointAddTagTransformer;
+import com.wavefront.agent.preprocessor.ReportPointBlacklistRegexFilter;
+import com.wavefront.agent.preprocessor.ReportPointDropTagTransformer;
+import com.wavefront.agent.preprocessor.ReportPointRenameTagTransformer;
+import com.wavefront.agent.preprocessor.ReportPointReplaceRegexTransformer;
+import com.wavefront.agent.preprocessor.ReportPointWhitelistRegexFilter;
+import com.wavefront.ingester.GraphiteDecoder;
+
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+
+import static org.junit.Assert.*;
+
+import sunnylabs.report.ReportPoint;
+
+public class PreprocessorRulesTest {
+
+  private static AgentPreprocessorConfiguration config;
+  private final static List<String> emptyCustomSourceTags = Collections.emptyList();
+  private final GraphiteDecoder decoder = new GraphiteDecoder(emptyCustomSourceTags);
+
+  @BeforeClass
+  public static void setup() throws IOException {
+    InputStream stream = PreprocessorRulesTest.class.getResourceAsStream("preprocessor_rules.yaml");
+    config = new AgentPreprocessorConfiguration();
+    config.loadFromStream(stream);
+  }
+
+  @Test(expected = NullPointerException.class)
+  public void testInvalidPointLineRule1() {
```

Comment:
if you want to have one test case for each invalid argument, i'd give each test case a more descriptive name, e.g.

> testNullRegexReplaceRuleThrowsFor...

If you don't care about such descriptiveness, I'd put all of the test cases in the same method, and use something like http://blog.codeleak.pl/2015/04/junit-testing-exceptions-with-java-8.html or https://code.google.com/archive/p/catch-exception/